### DISCUSSION TITLE
Relation help

### DISCUSSION THREAD

**Discussion by cduytran** [pocketbase/pocketbase#1810] [Q&A]
tickets with relation field (ticketPackRef) which references packs.
packs also has a relation field (packGameRef) which references games

how do I  run a query to get a ticket with both packs and games expanded?

await pb.collection('tickets').getFirstListItem(`ticketPackRef.packNumber = "${pack.packNumber}"
      && ticketNumber = ${ticketNumber} `,{
        expand:`ticketPackRef`
      })

Above code only returns with packs expanded.. how do I expand games? Thanks.

**Comment by ganigeorgiev** [comment]
If I'm understanding correctly the described structure, you should be able to use `ticketPackRef.packGameRef` as expand, eg.:

```js
await pb.collection('tickets').getFirstListItem(`ticketPackRef.packNumber = "${pack.packNumber}" && ticketNumber = ${ticketNumber}`, {
    expand: 'ticketPackRef.packGameRef'
})
```

_Note that the same expand rules applies for the nested relations too so you have to make sure that your request client satisfy the "View API" rule access of each relation._

**Reply by cduytran** [reply]
const ticket = serializeNonPOJOs( 
      await pb.collection('tickets').getFirstListItem(`ticketPackRef.packNumber = "${pack.packNumber}"
      && ticketNumber = ${ticketNumber} `,{
        expand:`ticketPackRef, ticketPackRef.packGameRef`
      }) 
    )

will produce result below...It doesn't include game in the result

ticket {
  collectionId: 'u4fm2bta46bwsew',
  collectionName: 'tickets',
  created: '2023-02-08 07:37:43.223Z',
  expand: { ticketPackRef: [ [Object] ] },
  id: 'itmhw36oatg0q9z',
  ticketNumber: 0,
  ticketPackRef: [ '1402v226lrc2hvx' ],
  ticketRecDate: '',
  ticketRetDate: '',
  ticketSoldDate: '',
  updated: '2023-02-08 07:37:43.223Z'
}

Not sure if I have the wrong syntax for chaining expand.

**Reply by ganigeorgiev** [reply]
The nested `packGameRef` relation will be nested in the expand of each `ticketPackRef` record (if exist).

Or in other words, it should be available in the `ticket.expand.ticketPackRef.expand` object (_assuming that the relations are single and not array_).
